src/logcollector/logcollector.c

```diff
@@ -782,7 +782,7 @@ int check_pattern_expand() {
                 continue;
             }
             while (g.gl_pathv[glob_offset] != NULL) {
-                if (current_files >= maximum_files) {
+                if (maximum_files && current_files >= maximum_files) {
```

Comment:
Is this check necessary? `maximum_files` should never be `0` in Logcollector, and it if was, it would be an error.